server/matchmaker/map_pool.py

```diff
@@ -0,0 +1,50 @@
+import random
+from collections import Counter
+from typing import List
+
+from ..decorators import with_logger
+from ..types import Map
+
+
+@with_logger
+class MapPool(object):
+    def __init__(
+        self,
+        map_pool_id: int,
+        name: str,
+        maps: List[Map] = []
+    ):
+        self.id = map_pool_id
+        self.name = name
+        self.set_maps(maps)
+
+    def set_maps(self, maps: List[Map]) -> None:
+        self.maps = {map_.id: map_ for map_ in maps}
+
+    def choose_map(self, played_map_ids: List[int] = []) -> Map:
+        """
+        Select a random map who's id does not appear in `played_map_ids`. If
+        all map ids appear in the list, then pick one that appears the least
+        amount of times.
+        """
+        if not self.maps:
+            self._logger.critical(
+                "Trying to choose a map from an empty map pool: %s", self.name
+            )
+            raise RuntimeError(f"Map pool {self.name} not set!")
+
+        # Make sure the counter has every available map
+        counter = Counter(self.maps.keys())
```

Comment:
If you wanted to avoid adding one to every count, you could use
```python
counter = Counter({map: 0 for map in self.maps.keys()})
```
which might make it more future-proof (but adds one dict-comp).

Reply:
I did that at first, but then I realized that I don't care what the actual count is. Feels nicer to me to have less code and no intermediate data structure.

Reply:
Makes sense!